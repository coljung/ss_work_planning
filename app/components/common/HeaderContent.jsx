import React from 'react';
import Logo from './Logo';
// import NavigationUser from './NavigationUser';

export default () =>
  <div className="headerContent">
     <Logo />
     <h1>Merchandise Planning Tool</h1>
     {/* <NavigationUser /> */}
 </div>
