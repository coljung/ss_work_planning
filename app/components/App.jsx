import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Layout, Icon } from 'antd';
import { Content, Header, Sider } from 'antd/lib/layout';
import HeaderContent from './common/HeaderContent';
import CustomNavigation from './customNavigation/CustomNavigation';
import { NotificationManager } from '../notifications/NotificationManager';

export default class App extends Component {
    constructor(props) {
        super(props);
        this.state = { collapsed: true, showStoreModal: true };
    }

    toggle() {
        const collapsed = !this.state.collapsed;
        this.setState({ collapsed });
        clearTimeout(this.timer);

        // collapse after 8 seconds
        if (!collapsed) {
            this.timer = setTimeout(() => {
                this.setState({ collapsed: !collapsed });
            }, 80000);
        }
    }

    render() {
        return (
            <div className="store_layout">
                <Header>
                    <Icon
                        className="trigger"
                        type={this.state.collapsed ? 'menu-unfold' : 'menu-fold'}
                        onClick={ this.toggle.bind(this) } />
                        <HeaderContent />
                </Header>
                <Layout>
                    <Sider
                        trigger={null}
                        collapsible
                        collapsed={this.state.collapsed}>
                        <CustomNavigation pathname={this.props.location.pathname} />
                    </Sider>
                    <Content>
                        <main style={{ flex: 1, overflowY: 'auto', padding: '0 25px 25px' }}>
                            {this.props.children}
                            <NotificationManager />
                        </main>
                    </Content>
                </Layout>
            </div>
        );
    }
}

App.propTypes = {
    location: PropTypes.object,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.element),
        PropTypes.element,
    ]),
};
