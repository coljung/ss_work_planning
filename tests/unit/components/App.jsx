jest.mock('../../../app/components/common/NavigationMain', () => 'NavigationMain');
jest.mock('../../../app/notifications/NotificationManager', () => 'NotificationManager12345643');

import React from 'react';
import { shallow } from 'enzyme';
import TestRenderer from 'react-test-renderer';
import App from '../../../app/components/App';
import HeaderContent from '../../../app/components/common/HeaderContent';

describe('App', () => {
    it('should render correctly', () => {
        const testRenderer = TestRenderer.create(
            <App  location={{pathname: 'pathname'}} />
        );
        const wrapper = shallow(
            <App  location={{pathname: 'pathname'}} />
        );
        expect(testRenderer).toMatchSnapshot()
	});
})
