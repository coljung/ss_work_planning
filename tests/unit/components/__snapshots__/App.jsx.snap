// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="app_layout"
>
  <div
    className="ant-layout ant-layout-has-sider"
  >
    <div
      className="ant-layout-header"
    >
      <i
        className="anticon anticon-menu-unfold trigger"
        onClick={[Function]}
      />
      <div
        className="headerContent"
      >
        <svg
          viewBox="0 0 540 110"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M482.4,90.6V59.8h48V47.5h-48V19.4h56.2V7h-69.9v96H540V90.6H482.4z M340.4,86.4L295.8,7h-19.2v96h13.7V23.6l44.6,79.4h19.2 V7h-13.7L340.4,86.4z M199.2,59.8h48V47.5h-48V19.4h56.2V7h-69.9v96h71.3V90.6h-57.6V59.8z M411.8,18h1.4c11.7,0,20.4,6.9,21.1,17.8 H448c-0.7-18.5-14.3-30.2-35.5-30.2c-20.7,0-34.3,12.3-34.3,27.4s8.9,23.9,24.3,27l15.4,3.2c10.1,2.1,15.1,6.6,15.1,14.4 s-7,14.4-19.9,14.4h-1.4c-14.4,0-24.5-6.9-24.5-19.9h-13.7c0.7,21.2,15.1,32.2,37.7,32.2c21.2,0,35.5-10.4,35.5-27.3 c0-15.6-9.6-24.4-26.2-27.7l-15.4-3c-8.1-1.6-13.2-6-13.2-13.4C391.9,24.4,399.9,18,411.8,18 M128.5,18h1.4 c11.7,0,20.4,6.9,21.1,17.8h13.7C164,17.3,150.4,5.6,129.1,5.6c-20.7,0-34.3,12.3-34.3,27.4s8.9,23.9,24.3,27l15.4,3.2 c10.1,2.1,15.1,6.6,15.1,14.4s-7,14.4-19.9,14.4h-1.4c-14.4,0-24.5-6.9-24.5-19.9H90.1c0.7,21.2,15.1,32.2,37.7,32.2 c21.2,0,35.5-10.4,35.5-27.3c0-15.6-9.6-24.4-26.2-27.7l-15.4-3c-8.1-1.6-13.2-6-13.2-13.4C108.6,24.4,116.5,18,128.5,18 M38.4,18 h1.4c11.7,0,20.4,6.9,21.1,17.8h13.7C73.9,17.3,60.3,5.6,39.1,5.6C18.4,5.6,4.8,18,4.8,33.1s8.9,23.9,24.3,27l15.4,3.2 c10.1,2.1,15.1,6.6,15.1,14.4S52.5,92,39.6,92h-1.4c-14.4,0-24.5-6.9-24.5-19.9H0c0.7,21.2,15.1,32.2,37.7,32.2 c21.2,0,35.5-10.4,35.5-27.3c0-15.6-9.6-24.4-26.2-27.7l-15.4-3c-8.1-1.6-13.2-6-13.2-13.4C18.5,24.4,26.5,18,38.4,18"
          />
        </svg>
        <h1>
          Merchandise Planning Tool
        </h1>
      </div>
    </div>
    <div
      className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed"
      style={
        Object {
          "flex": "0 0 80px",
          "maxWidth": "80px",
          "minWidth": "80px",
          "width": "80px",
        }
      }
    >
      <div
        className="ant-layout-sider-children"
      >
        <CustomNavigation
          pathname="pathname"
          triggerMenuCollapse={[Function]}
        />
      </div>
    </div>
    <div
      className="ant-layout-content"
    >
      <main
        style={
          Object {
            "flex": 1,
            "overflowY": "auto",
            "padding": "0 25px 25px",
          }
        }
      >
        <NotificationManager12345643 />
      </main>
    </div>
  </div>
</div>
`;
